src/PreferencesMatrixGenerator.java

```diff
@@ -0,0 +1,210 @@
+    // Example: PreferencesMatrixGenerator 10 0 "test1.txt"
+    public static void main(String args[])
+    {
+        final int kExpectedNumberOfArguments = 3;
+
+        if (kExpectedNumberOfArguments == args.length)
+        {
+            final int kDataSetSize = Integer.parseInt(args[0]);
+            final int kMode = Integer.parseInt(args[1]);
+            final String kFileName = args[2];
+
+            PreferenceGeneratorMode mode = PreferenceGeneratorMode.kBestCase;
+
+            switch(kMode)
+            {
+                case 0:
+                    mode = PreferenceGeneratorMode.kBestCase;
+                    break;
+                case 1:
+                    mode = PreferenceGeneratorMode.kWorstCase;
+                    break;
+                case 2:
+                    mode = PreferenceGeneratorMode.kRandom;
+                    break;
+                default:
+                    System.out.println("Invalid Mode!");
+                    System.out.println("Valid Modes: BestCase:0 WorstCase:1 Random:2");
+                    System.exit(-100);
+            }
+
+            PreferencesMatrixGenerator preferencesMatrixGenerator = new PreferencesMatrixGenerator(kDataSetSize, mode);
+            //preferencesMatrixGenerator.printItems();
+            preferencesMatrixGenerator.writeToFile(kFileName);
+        }
+        else
+        {
+            System.out.println("Invalid Number of Arguments");
+            System.out.println("Usage: PreferencesMatrixGenerator NumberOfPreferences Mode \"outputFile.txt\"");
+            System.out.println("Example: PreferencesMatrixGenerator 100 0 \"test1.txt\"");
+            System.out.println("Valid Modes: BestCase:0 WorstCase:1 Random:2");
+            System.exit(-100);
+        }
+    }
+
+    private int m_dataSize;
+    private int[][] m_menPreferenceMatrix;
+    private int[][] m_womenPreferenceMatrix;
+    private PreferenceGeneratorMode m_mode;
+
+    PreferencesMatrixGenerator(int dataSize, PreferenceGeneratorMode mode)
+    {
+        m_dataSize = dataSize;
+        m_mode = mode;
+        m_menPreferenceMatrix = new int[m_dataSize][m_dataSize];
+        m_womenPreferenceMatrix = new int[m_dataSize][m_dataSize];
+
+        populatePreferenceMatrices();
+    }
+
+    private void populatePreferenceMatrices()
+    {
+        switch (m_mode)
+        {
+
+            case kBestCase:
+                // Best case is when men and women have the same preference lists
+                for (int i = 0; i < m_dataSize; i++)
+                {
+                    for (int j = 0 ; j < m_dataSize; j++)
+                    {
+                        m_menPreferenceMatrix[i][j] = j + 1;
+                        m_womenPreferenceMatrix[i][j] = j + 1;
+                    }
+                }
+                break;
+
+            case kWorstCase:
+                // Worst case is when men and women have inverted preference lists
```

Comment:
According to [S. S. Tseng and R. C. T. Lee](https://ir.nctu.edu.tw/bitstream/11536/4865/1/A1984TX77500006.pdf) they say that following McVitie and Wilson algorithm a worst case for a 8x8 is:
```
M1 1 7 6 5 4 3 2 8   W1 7 6 5 4 3 2 1 8 
M2 2 1 7 6 5 4 3 8   W2 1 8 7 6 5 4 3 2 
M3 3 2 1 7 6 5 4 8   W3 2 1 8 7 6 5 4 3 
M4 4 3 2 1 7 6 5 8   W4 3 2 1 8 7 6 5 4 
M5 5 4 3 2 1 7 6 8   W5 4 3 2 1 8 7 6 5 
M6 6 5 4 3 2 1 7 8   W6 5 4 3 2 1 8 7 6 
M7 7 6 5 4 3 2 1 8   W7 6 5 4 3 2 1 8 7 
M8 1 7 6 5 4 3 2 8   W8 8 7 6 5 4 3 2 1 
```
Do you think is the same case for Gale-Shapley? Doesn't seem like just inverting the preference list will do it. Although maybe is not critical to come up with the worst case right now.

Reply:
You are right, what you gave is a worst case. I also added one that is smaller so it is easy to see how it is a worst case. I added a debug check to know whether an input file is a worst case or not.